

<!DOCTYPE html>
<html class="writer-html5" lang="en" data-content_root="../">
<head>
  <meta charset="utf-8" />
  <meta name="readthedocs-addons-api-version" content="1"><meta name="viewport" content="width=device-width, initial-scale=1" />

    <!-- Google tag (gtag.js) -->
    <script async src="https://www.googletagmanager.com/gtag/js?id=G-6GH9ZZT8G3"></script>
    <script>
        window.dataLayer = window.dataLayer || [];
        function gtag(){dataLayer.push(arguments);}
        gtag('js', new Date());
        gtag('config', 'G-6GH9ZZT8G3');
    </script>
    
  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Local Deployment &mdash; GenPipes  6.0.0
 documentation</title>
      <link rel="stylesheet" type="text/css" href="../_static/pygments.css?v=b86133f3" />
      <link rel="stylesheet" type="text/css" href="../_static/css/theme.css?v=e59714d7" />
      <link rel="stylesheet" type="text/css" href="../_static/togglebutton.css?v=13237357" />
      <link rel="stylesheet" type="text/css" href="../_static/sphinx-design.min.css?v=95c83b7e" />
      <link rel="stylesheet" type="text/css" href="../_static/css/custom.css?v=f95d828e" />

  
      <script src="../_static/jquery.js?v=5d32c60e"></script>
      <script src="../_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
      <script src="../_static/documentation_options.js?v=8ccdf785"></script>
      <script src="../_static/doctools.js?v=9bcbadda"></script>
      <script src="../_static/sphinx_highlight.js?v=dc90522c"></script>
      <script>let toggleHintShow = 'Click to show';</script>
      <script>let toggleHintHide = 'Click to hide';</script>
      <script>let toggleOpenOnPrint = 'true';</script>
      <script src="../_static/togglebutton.js?v=4a39c7ea"></script>
      <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
      <script src="../_static/design-tabs.js?v=f930bc37"></script>
      <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script src="../_static/js/theme.js"></script>
    <script src="../_static/js/versions.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Container Deployment" href="dep_gp_container.html" />
    <link rel="prev" title="DRAC Deployment" href="access_gp_pre_installed.html" /> 
<script async type="text/javascript" src="/_/static/javascript/readthedocs-addons.js"></script><meta name="readthedocs-project-slug" content="genpipes" /><meta name="readthedocs-version-slug" content="genpipes-v6.0.0" /><meta name="readthedocs-resolver-filename" content="/deploy/dep_gp_local.html" /><meta name="readthedocs-http-status" content="200" /></head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search"  style="background: #1c75bc" >

          
          
          <a href="../index.html" class="icon icon-home">
            GenPipes
              <img src="../_static/genpipes_doc_img.png" class="logo" alt="Logo"/>
          </a>
              <div class="switch-menus">
                <div class="version-switch"></div>
                <div class="language-switch"></div>
              </div>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">GenPipes</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../about/index.html">About</a></li>
<li class="toctree-l1"><a class="reference internal" href="../get-started/gp_usecases.html">Use Cases</a></li>
<li class="toctree-l1"><a class="reference internal" href="../faq/gp_faq.html">FAQ</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Get Started</span></p>
<ul class="current">
<li class="toctree-l1 current"><a class="reference internal" href="how_to_deploy_genpipes.html">Deploying GenPipes</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="dep_options.html">Deployment Options</a></li>
<li class="toctree-l2"><a class="reference internal" href="access_gp_pre_installed.html">DRAC Deployment</a></li>
<li class="toctree-l2 current"><a class="current reference internal" href="#">Local Deployment</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#step-1-download-latest-genpipes-sources-and-install-with-pip">Step 1: Download latest GenPipes sources and install with pip</a></li>
<li class="toctree-l3"><a class="reference internal" href="#step-2-setup-environment-variables">Step 2: Setup environment variables</a></li>
<li class="toctree-l3"><a class="reference internal" href="#step-3-accessing-software-modules-and-genomes-needed-for-genpipe">Step 3: Accessing software modules and genomes needed for GenPipe</a></li>
<li class="toctree-l3"><a class="reference internal" href="#step-4-validating-genpipes-local-deployment">Step 4: Validating GenPipes local deployment</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="dep_gp_container.html">Container Deployment</a></li>
<li class="toctree-l2"><a class="reference internal" href="dep_gp_cloud_gcp.html">Cloud Deployment</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../get-started/index.html">Using GenPipes</a></li>
<li class="toctree-l1"><a class="reference internal" href="../user_guide/user_guide.html">User Guide</a></li>
<li class="toctree-l1"><a class="reference internal" href="../tutorials/list_tutorials.html">Tutorials</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Support</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../support/how_to_get_support.html">Need Help?</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Community</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../community/channels.html">Channels</a></li>
<li class="toctree-l1"><a class="reference internal" href="../community/contributing.html">Contributions</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Resources</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../resources/citation.html">Citations</a></li>
<li class="toctree-l1"><a class="reference internal" href="../resources/publications.html">Publications</a></li>
<li class="toctree-l1"><a class="reference internal" href="../resources/workshops.html">Workshops</a></li>
<li class="toctree-l1"><a class="reference internal" href="../resources/testdataset.html">Test Datasets</a></li>
<li class="toctree-l1"><a class="reference internal" href="../resources/compute_resources.html">C3G Resources</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Development</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../development/release_notes.html">Release Notes</a></li>
<li class="toctree-l1"><a class="reference internal" href="../development/gp_release_instructions.html">Release Instructions</a></li>
<li class="toctree-l1"><a class="reference internal" href="../development/dev_guide.html">Developer Guide</a></li>
<li class="toctree-l1"><a class="reference internal" href="../development/troubleshooting_guide.html">Troubleshooting Guide</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Documentation</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../documentation/about.html">About Docs</a></li>
<li class="toctree-l1"><a class="reference internal" href="../documentation/genpipes_doc_archmap.html">Site Map</a></li>
<li class="toctree-l1"><a class="reference internal" href="../documentation/docs_changelog.html">Changelog</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu"  style="background: #1c75bc" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">GenPipes</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home" aria-label="Home"></a></li>
          <li class="breadcrumb-item"><a href="how_to_deploy_genpipes.html">Deploying GenPipes</a></li>
      <li class="breadcrumb-item active">Local Deployment</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../_sources/deploy/dep_gp_local.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="local-deployment">
<span id="docs-dep-gp-local"></span><h1>Local Deployment<a class="headerlink" href="#local-deployment" title="Link to this heading">ÔÉÅ</a></h1>
<p>This document explains how to deploy GenPipes locally on a bare metal or virtual server. See <a class="reference internal" href="dep_gp_container.html#docs-dep-gp-container"><span class="std std-ref">GenPipes in a container</span></a> section to install GenPipes locally in a container.</p>
<p>Check out <a class="reference internal" href="dep_options.html#docs-dep-options"><span class="std std-ref">GenPipes Deployment Options Page</span></a> for other available options to use GenPipes.</p>
<section id="step-1-download-latest-genpipes-sources-and-install-with-pip">
<span id="docs-download-gp-src"></span><h2>Step 1: Download latest GenPipes sources and install with pip<a class="headerlink" href="#step-1-download-latest-genpipes-sources-and-install-with-pip" title="Link to this heading">ÔÉÅ</a></h2>
<p>First of all, visit GenPipes the <a class="reference external" href="https://github.com/c3g/GenPipes/releases/download/">Download Page</a> and get a copy of the latest stable release software.</p>
<p>Note that the repo was earlier hosted on Bitbucket until v5.1.0 of GenPipes.</p>
<div class="sd-tab-set docutils">
<input checked="checked" id="sd-tab-item-0" name="sd-tab-set-0" type="radio">
<label class="sd-tab-label" for="sd-tab-item-0">
v6.0</label><div class="sd-tab-content docutils">
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">git</span> <span class="n">clone</span> <span class="n">https</span><span class="p">:</span><span class="o">//</span><span class="n">github</span><span class="o">.</span><span class="n">com</span><span class="o">/</span><span class="n">c3g</span><span class="o">/</span><span class="n">GenPipes</span><span class="o">.</span><span class="n">git</span>
<span class="n">cd</span> <span class="n">GenPipes</span>
<span class="n">pip</span> <span class="n">install</span> <span class="o">.</span>
</pre></div>
</div>
</div>
<input id="sd-tab-item-1" name="sd-tab-set-0" type="radio">
<label class="sd-tab-label" for="sd-tab-item-1">
v5.x</label><div class="sd-tab-content docutils">
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">git</span> <span class="n">clone</span> <span class="n">git</span><span class="nd">@bitbucket</span><span class="o">.</span><span class="n">org</span><span class="p">:</span><span class="n">mugqic</span><span class="o">/</span><span class="n">genpipes</span><span class="o">.</span><span class="n">git</span>
<span class="n">cd</span> <span class="o">&lt;</span><span class="n">bitbucket_repo</span><span class="o">&gt;</span>
<span class="n">pip</span> <span class="n">install</span> <span class="o">.</span>
</pre></div>
</div>
</div>
</div>
</section>
<section id="step-2-setup-environment-variables">
<h2>Step 2: Setup environment variables<a class="headerlink" href="#step-2-setup-environment-variables" title="Link to this heading">ÔÉÅ</a></h2>
<p>Add the following lines in your your <em>$HOME/.bash_profile</em>: to set MUGQIC_PIPELINES_HOME to your local copy path. For example,</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>export MUGQIC_PIPELINES_HOME=/path/to/your/local/genpipes
export GENPIPES_INIS=$MUGQIC_PIPELINES_HOME/genpipes/pipelines
</pre></div>
</div>
</section>
<section id="step-3-accessing-software-modules-and-genomes-needed-for-genpipe">
<span id="accessing-sw-mod-genomes-local-dp"></span><h2>Step 3: Accessing software modules and genomes needed for GenPipe<a class="headerlink" href="#step-3-accessing-software-modules-and-genomes-needed-for-genpipe" title="Link to this heading">ÔÉÅ</a></h2>
<p>GenPipes was formerly known as MUGQIC Pipelines. Genomic analysis executed using these pipelines requires <a class="reference internal" href="../c3gres/cvmfs_genomes.html#doc-cvmfs-genomes"><span class="std std-ref">genomes</span></a> and <a class="reference internal" href="../c3gres/cvmfs_modules.html#doc-cvmfs-modules"><span class="std std-ref">software modules</span></a>. You need to load the software modules in your shell environment. To do so, set the environment variable <strong>MUGQIC_INSTALL_HOME</strong> to the directory where you want to install those resources in your <strong>$HOME/.bash_profile</strong> as follows:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>## MUGQIC genomes and modules

export MUGQIC_INSTALL_HOME=/path/to/your/local/genpipes/resources
module use $MUGQIC_INSTALL_HOME/modulefiles
</pre></div>
</div>
<p><strong>Installing available modules</strong></p>
<p>Software tools and associated modules must be installed in $MUGQIC_INSTALL_HOME/software/ and $MUGQIC_INSTALL_HOME/modulefiles/.  Default software/module installation scripts are already available in $MUGQIC_PIPELINES_HOME/resources/modules/.</p>
<p><strong>Install new modules</strong></p>
<p>To install a new module or new software tool and associated modules semi-automatically, use the following instructions:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>cp $MUGQIC_PIPELINES_HOME/resources/modules/MODULE_INSTALL_TEMPLATE.sh $MUGQIC_PIPELINES_HOME/resources/modules/&lt;my_software&gt;.sh
</pre></div>
</div>
<p>Follow the instructions in the file $MUGQIC_PIPELINES_HOME/resources/modules/&lt;my_software&gt;.sh and modify it accordingly.  Next you need to run the following command with <strong>No arguments</strong>. By default, it will download and extract the remote software archive, build the software and create the associated module, all in $MUGQIC_INSTALL_HOME_DEV if it is set.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$MUGQIC_PIPELINES_HOME/resources/modules/&lt;my_software&gt;.sh
</pre></div>
</div>
<p>If everything executes OK with no error, you are ready to install the <cite>my_software</cite> module in production. Use the command:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$MUGQIC_PIPELINES_HOME/resources/modules/&lt;my_software&gt;.sh MUGQIC_INSTALL_HOME
</pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Please note there is no $ before MUGQIC_INSTALL_HOME specified as argument above!.</p>
</div>
<p>Next, you need to check if the module is successfully installed and available for use by executing the following command:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">module</span> <span class="n">avail</span> <span class="mi">2</span><span class="o">&gt;&amp;</span><span class="mi">1</span> <span class="o">|</span> <span class="n">grep</span> <span class="n">mugqic</span><span class="o">/&lt;</span><span class="n">my_software</span><span class="o">&gt;/&lt;</span><span class="n">version</span><span class="o">&gt;</span>
</pre></div>
</div>
<p>This completes the software module setup for GenPipes execution. Next you need to make sure all required reference genomes are available in your local deployment. Refer to the next section if you wish to install additional genomes.</p>
<p id="ref-installing-genomes"><strong>Installing genomes</strong></p>
<p>Reference genomes and annotations must be installed in the following directory:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$MUGQIC_INSTALL_HOME/genomes/
</pre></div>
</div>
<p>Default genome installation scripts are already available locally in the following directory:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$MUGQIC_PIPELINES_HOME/resources/genomes/
</pre></div>
</div>
<p>To install all of the available genomes that are bundled with GenPipes package, use the following script:</p>
<p>$MUGQIC_PIPELINES_HOME/resources/genomes/install_all_genomes.sh</p>
<p>All species related files are in the following directory:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$MUGQIC_INSTALL_HOME/genomes/species/&lt;species_scientific_name&gt;.&lt;assembly&gt;/
</pre></div>
</div>
<p>For example, <em>Homo Sapiens</em> assembly <em>GRCh37</em> genome directory hierarchy is as follows:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$MUGQIC_INSTALL_HOME/genomes/species/Homo_sapiens.GRCh37/
‚îú‚îÄ‚îÄ annotations/
‚îÇ   ‚îú‚îÄ‚îÄ gtf_tophat_index/
‚îÇ   ‚îú‚îÄ‚îÄ Homo_sapiens.GRCh37.dbSNP142.vcf.gz
‚îÇ   ‚îú‚îÄ‚îÄ Homo_sapiens.GRCh37.dbSNP142.vcf.gz.tbi
‚îÇ   ‚îú‚îÄ‚îÄ Homo_sapiens.GRCh37.Ensembl75.geneid2Symbol.tsv
‚îÇ   ‚îú‚îÄ‚îÄ Homo_sapiens.GRCh37.Ensembl75.genes.length.tsv
‚îÇ   ‚îú‚îÄ‚îÄ Homo_sapiens.GRCh37.Ensembl75.genes.tsv
‚îÇ   ‚îú‚îÄ‚îÄ Homo_sapiens.GRCh37.Ensembl75.GO.tsv
‚îÇ   ‚îú‚îÄ‚îÄ Homo_sapiens.GRCh37.Ensembl75.gtf
‚îÇ   ‚îú‚îÄ‚îÄ Homo_sapiens.GRCh37.Ensembl75.ncrna.fa
‚îÇ   ‚îú‚îÄ‚îÄ Homo_sapiens.GRCh37.Ensembl75.rrna.fa
‚îÇ   ‚îú‚îÄ‚îÄ Homo_sapiens.GRCh37.Ensembl75.transcript_id.gtf
‚îÇ   ‚îú‚îÄ‚îÄ Homo_sapiens.GRCh37.Ensembl75.vcf.gz
‚îÇ   ‚îú‚îÄ‚îÄ ncrna_bwa_index/
‚îÇ   ‚îî‚îÄ‚îÄ rrna_bwa_index/
‚îú‚îÄ‚îÄ downloads/
‚îÇ   ‚îú‚îÄ‚îÄ ftp.1000genomes.ebi.ac.uk/
‚îÇ   ‚îú‚îÄ‚îÄ ftp.ensembl.org/
‚îÇ   ‚îî‚îÄ‚îÄ ftp.ncbi.nih.gov/
‚îú‚îÄ‚îÄ genome/
‚îÇ   ‚îú‚îÄ‚îÄ bowtie2_index/
‚îÇ   ‚îú‚îÄ‚îÄ bwa_index/
‚îÇ   ‚îú‚îÄ‚îÄ Homo_sapiens.GRCh37.dict
‚îÇ   ‚îú‚îÄ‚îÄ Homo_sapiens.GRCh37.fa
‚îÇ   ‚îú‚îÄ‚îÄ Homo_sapiens.GRCh37.fa.fai
‚îÇ   ‚îî‚îÄ‚îÄ star_index/
‚îú‚îÄ‚îÄ Homo_sapiens.GRCh37.ini
‚îî‚îÄ‚îÄ log/
</pre></div>
</div>
<p>The assembly name is the one used by the download source. For e.g. ‚ÄúGRCh37‚Äù is used for <a class="reference external" href="http://www.ensembl.org/">Ensembl</a>.</p>
<p>Each species directory contains a ‚Äú.ini‚Äù file such as:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="o">&lt;</span><span class="n">scientific_name</span><span class="o">&gt;.&lt;</span><span class="n">assembly</span><span class="o">&gt;.</span><span class="n">ini</span>
</pre></div>
</div>
<p>Among other things, this ‚Äú.ini‚Äù file lists the assembly synonyms. In case of ‚Äúhg19‚Äù, the contents of Homo_sapiens.GRCh37.ini are as shown below:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="p">[</span><span class="n">DEFAULT</span><span class="p">]</span>
<span class="n">scientific_name</span><span class="o">=</span><span class="n">Homo_sapiens</span>
<span class="n">common_name</span><span class="o">=</span><span class="n">Human</span>
<span class="n">assembly</span><span class="o">=</span><span class="n">GRCh37</span>
<span class="n">assembly_synonyms</span><span class="o">=</span><span class="n">hg19</span>
<span class="n">source</span><span class="o">=</span><span class="n">Ensembl</span>
<span class="n">version</span><span class="o">=</span><span class="mi">75</span>
<span class="n">dbsnp_version</span><span class="o">=</span><span class="mi">142</span>
</pre></div>
</div>
<p><strong>Install a new Genome</strong></p>
<p>New genomes and annotations can be installed semi-automatically from <a class="reference external" href="http://www.ensembl.org/">Ensembl</a> (vertebrate species), <a class="reference external" href="http://ensemblgenomes.org">Ensemble Genomes</a> (other species) or <a class="reference external" href="http://genome.ucsc.edu/">UCSC</a> (genome and indexes only; no annotations).</p>
<p><em>Example - how to set up genomes for Chimpanzee:</em></p>
<ol class="arabic simple">
<li><p>Retrieve the species scientific name on <a class="reference external" href="http://ensemblgenomes.org">Ensemble Genomes</a> or <a class="reference external" href="http://genome.ucsc.edu/">UCSC</a> :</p></li>
</ol>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">Pan</span> <span class="n">troglodytes</span>
</pre></div>
</div>
<ol class="arabic simple" start="2">
<li><p>Retrieve the assembly name:</p>
<ul class="simple">
<li><p>Ensembl: ‚ÄúCHIMP2.1.4‚Äù</p></li>
<li><p>UCSC: ‚ÄúpanTro4‚Äù</p></li>
</ul>
</li>
<li><p>Retrieve the source version:</p>
<ul class="simple">
<li><p>Ensembl: ‚Äú78‚Äù</p></li>
<li><p>UCSC: unfortunately, UCSC does not have version numbers. Use <a class="reference external" href="http://hgdownload.soe.ucsc.edu/goldenPath/panTro4/bigZips/">panTro4.2bit</a> date formatted as ‚ÄúYYYY-MM-DD‚Äù: ‚Äú2012-01-09‚Äù</p></li>
</ul>
</li>
<li><p>Next, copy the template file to a new file name using the scientific name.</p></li>
</ol>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>cp $MUGQIC_PIPELINES_HOME/resources/genomes/GENOME_INSTALL_TEMPLATE.sh $MUGQIC_PIPELINES_HOME/resources/genomes/&lt;scientific_name&gt;.&lt;assembly&gt;.sh
</pre></div>
</div>
<p>For example, in case of Ensembl, use the following command:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>cp $MUGQIC_PIPELINES_HOME/resources/genomes/GENOME_INSTALL_TEMPLATE.sh $MUGQIC_PIPELINES_HOME/resources/genomes/Pan_troglodytes.CHIMP2.1.4.sh
</pre></div>
</div>
<p>In case of genomes from UCSC, use the following command to copy the genome install instructions:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>cp $MUGQIC_PIPELINES_HOME/resources/genomes/GENOME_INSTALL_TEMPLATE.sh $MUGQIC_PIPELINES_HOME/resources/genomes/Pan_troglodytes.panTro4.sh
</pre></div>
</div>
<ol class="arabic simple" start="5">
<li><p>Next, you need to modify the following file:</p></li>
</ol>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$MUGQIC_PIPELINES_HOME/resources/genomes/&lt;scientific_name&gt;.&lt;assembly&gt;.sh
</pre></div>
</div>
<p>Please note that ASSEMBLY_SYNONYMS can be left empty but if you know that 2 assemblies
are identical apart from chr sequence prefixes, document it.</p>
<p>Example below shows the modifications for Ensembl:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">SPECIES</span><span class="o">=</span><span class="n">Pan_troglodytes</span>   <span class="c1"># With &quot;_&quot;; no space!</span>
<span class="n">COMMON_NAME</span><span class="o">=</span><span class="n">Chimpanzee</span>
<span class="n">ASSEMBLY</span><span class="o">=</span><span class="n">CHIMP2</span><span class="mf">.1.4</span>
<span class="n">ASSEMBLY_SYNONYMS</span><span class="o">=</span><span class="n">panTro4</span>
<span class="n">SOURCE</span><span class="o">=</span><span class="n">Ensembl</span>
<span class="n">VERSION</span><span class="o">=</span><span class="mi">78</span>
</pre></div>
</div>
<p>Example below shows the modifications for UCSC:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">SPECIES</span><span class="o">=</span><span class="n">Pan_troglodytes</span>   <span class="c1"># With &quot;_&quot;; no space!</span>
<span class="n">COMMON_NAME</span><span class="o">=</span><span class="n">Chimpanzee</span>
<span class="n">ASSEMBLY</span><span class="o">=</span><span class="n">panTro4</span>
<span class="n">ASSEMBLY_SYNONYMS</span><span class="o">=</span><span class="n">CHIMP2</span><span class="mf">.1.4</span>
<span class="n">SOURCE</span><span class="o">=</span><span class="n">UCSC</span>
<span class="n">VERSION</span><span class="o">=</span><span class="mi">2012</span><span class="o">-</span><span class="mi">01</span><span class="o">-</span><span class="mi">09</span>
</pre></div>
</div>
<ol class="arabic simple" start="6">
<li><p>Now you can run the following command to install the genome in $MUGQIC_INSTALL_HOME_DEV (by default). This will download and install genomes, indexes and, for Ensembl only, annotations (GTF, VCF, etc.).</p></li>
</ol>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>bash $MUGQIC_PIPELINES_HOME/resources/genomes/&lt;scientific_name&gt;.&lt;assembly&gt;.sh
</pre></div>
</div>
<p><strong>Admin-only</strong>
To install it in $MUGQIC_INSTALL_HOME, run the following command:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>bash $MUGQIC_PIPELINES_HOME/resources/genomes/&lt;scientific_name&gt;.&lt;assembly&gt;.sh MUGQIC_INSTALL_HOME
</pre></div>
</div>
<ol class="arabic simple" start="7">
<li><p><strong>Admin-only</strong> If the new genome has been installed in $MUGQIC_INSTALL_HOME_DEV, to deploy in $MUGQIC_INSTALL_HOME you can use the following command:</p></li>
</ol>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>rsync -vca --no-o --no-g --no-p --size-only -I -O --ignore-times $MUGQIC_INSTALL_HOME_DEV/genomes/species/&lt;scientific_name&gt;.&lt;assembly&gt; $MUGQIC_INSTALL_HOME/genomes/species/
</pre></div>
</div>
<ol class="arabic simple" start="8">
<li><p>Lastly, add the newly created ‚Äú.ini‚Äù file to the genome configuration files for further use in subsequent genomic analysis pipeline runs by the following command:</p></li>
</ol>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>cp $MUGQIC_INSTALL_HOME/genomes/species/&lt;scientific_name&gt;.&lt;assembly&gt;/&lt;scientific_name&gt;.&lt;assembly&gt;.ini $MUGQIC_PIPELINES_HOME/resources/genomes/config/
</pre></div>
</div>
</section>
<section id="step-4-validating-genpipes-local-deployment">
<h2>Step 4: Validating GenPipes local deployment<a class="headerlink" href="#step-4-validating-genpipes-local-deployment" title="Link to this heading">ÔÉÅ</a></h2>
<p>You are now all set to use GenPipes pipelines. For each pipeline, you can get help about its usage through the help command:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">genpipes</span> <span class="o">&lt;</span><span class="n">pipeline_name</span><span class="o">&gt;</span> <span class="o">--</span><span class="n">help</span>
</pre></div>
</div>
<p>Running pipelines requires other inputs such as <a class="reference internal" href="../get-started/concepts/config_ini_file.html#docs-config-ini-file"><span class="std std-ref">Configuration File</span></a>, <a class="reference internal" href="../get-started/concepts/readset_file.html#docs-readset-file"><span class="std std-ref">Readset File</span></a> and <a class="reference internal" href="../get-started/concepts/design_file.html#docs-design-file"><span class="std std-ref">Design File</span></a>. For details on how to run individual pipelines you can see <a class="reference internal" href="../get-started/using_gp.html#docs-using-gp"><span class="std std-ref">Running GenPipes</span></a> or <a class="reference internal" href="../user_guide/user_guide.html#docs-user-guide"><span class="std std-ref">GenPipes User Guide</span></a>.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>In case of any issues, you can try GenPipes <a class="reference internal" href="../support/how_to_get_support.html#docs-how-to-get-support"><span class="std std-ref">Support</span></a> or check out other <a class="reference internal" href="../community/channels.html#docs-channels"><span class="std std-ref">communication channels</span></a> to view latest discussions around using GenPipes by the community.</p>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>You may also want to check the latest GenPipes deployment and setup instructions listed in <a class="reference external" href="https://github.com/c3g/GenPipes/blob/main/README.md">GenPipes README.md file</a>.</p>
</div>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="access_gp_pre_installed.html" class="btn btn-neutral float-left" title="DRAC Deployment" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="dep_gp_container.html" class="btn btn-neutral float-right" title="Container Deployment" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright GenPipes (GSoD 2019-25).
      <span class="lastupdated">Last updated on 
 Aug 15, 2025 at 05:10 by shaloo.
      </span></p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>